tensorflow/lite/kernels/sparse_to_dense.cc

```diff
@@ -202,17 +188,20 @@ TfLiteStatus SparseToDenseImpl(TfLiteContext* context, TfLiteNode* node) {
                       ResizeOutputShape(context, output_shape, output));
   }
 
-  const int num_indices = SizeOfDimension(indices, 0);
+  const int num_indices =
+      NumDimensions(indices) == 0 ? 1 : SizeOfDimension(indices, 0);
+  const int true_dim =
+      NumDimensions(indices) == 2 ? SizeOfDimension(indices, 1) : 1;
   const bool value_is_scalar = NumDimensions(values) == 0;
-  std::vector<std::vector<TI>> indices_vector;
-  indices_vector.reserve(num_indices);
+  TI* indices_vector = (TI*)malloc(num_indices * true_dim * sizeof(TI));
```

Comment:
a one-dimensional std::vector is fine here, and preferred over malloc.

Reply:
@jdduke : Thanks for your feedback! You are right it is more preferable. But i have used malloc for 2 reasons:
   1:> The scope for indices_vector is only for 1 function which need only read-only behavior.
   2:> Some times the indices size can be more, in that case it will help save some memory.
Please let me know your valuable opinion. Thanks!

Reply:
We should still prefer std::vector. An alternative would be std::unique_ptr<TI[]>.